lib/requestwrapper.ts

```diff
@@ -179,59 +156,111 @@ export function sendRequest(parameters, _callback) {
-    );
+    Object.keys(formData).forEach(key => {
+      if (formData[key].data != null) {
+        formData[key] = buildRequestFileObject(formData[key]);
+      }
+    });
 
     // Stringify arrays
-    Object.keys(formData).forEach(
-      key => Array.isArray(formData[key]) && (formData[key] = formData[key].join(','))
-    );
+    Object.keys(formData).forEach(key => {
+      if (Array.isArray(formData[key])) {
+        formData[key] = formData[key].join(',');
+      }
+    });
 
     // Convert non-file form parameters to strings
-    Object.keys(formData).forEach(
-      key =>
-        !isFileParam(formData[key]) &&
+    Object.keys(formData).forEach(key => {
+      if (!isFileParam(formData[key]) &&
         !Array.isArray(formData[key]) &&
-        typeof formData[key] === 'object' &&
-        (formData[key] = JSON.stringify(formData[key]))
-    );
+        typeof formData[key] === 'object') {
+        (formData[key] = JSON.stringify(formData[key]));
+      }
+    });
+
+    // build multipart form data
+    Object.keys(formData).forEach(key => {
+      // handle files differently to maintain options
+      if (formData[key].value) {
+        multipartForm.append(key, formData[key].value, formData[key].options);
+      } else {
+        multipartForm.append(key, formData[key]);
+      }
+    });
   }
 
   // Path params
-  options.url = parsePath(options.url, path);
-  delete options.path;
+  url = parsePath(url, path);
 
   // Headers
-  options.headers = extend({}, options.headers);
+  headers = extend({}, headers);
   if (!isBrowser) {
-    options.headers['User-Agent'] = `${pkg.name}-nodejs-${pkg.version};${options.headers[
-      'User-Agent'
-    ] || ''}`;
+    headers['User-Agent'] = `${pkg.name}-nodejs-${pkg.version};${headers['User-Agent'] || ''}`;
   }
 
-  // Query params
-  if (options.qs && Object.keys(options.qs).length > 0) {
-    Object.keys(options.qs).forEach(
-      key => Array.isArray(options.qs[key]) && (options.qs[key] = options.qs[key].join(','))
+  // Convert array-valued query params to strings
+  if (qs && Object.keys(qs).length > 0) {
+    Object.keys(qs).forEach(
+      key => Array.isArray(qs[key]) && (qs[key] = qs[key].join(','))
     );
-    options.useQuerystring = true;
   }
 
   // Add service default endpoint if options.url start with /
-  if (options.url.charAt(0) === '/') {
-    options.url = parameters.defaultOptions.url + options.url;
+  if (url && url.charAt(0) === '/') {
+    url = parameters.defaultOptions.url + url;
+  }
+
+  let data = body;
+
+  if (form) {
+    data = querystring.stringify(form);
+    headers['Content-type'] = 'application/x-www-form-urlencoded';
+  }
+
+  if (formData) {
+    data = multipartForm;
+    // form-data generates headers that MUST be included or the request will fail
+    headers = extend(true, {}, headers, multipartForm.getHeaders());
   }
 
-  // Compression support
-  options.gzip = true;
+  const axiosObject = {
```

Comment:
I would call it `requestParams`. I don't think we need `axios` to be part of the name